lib/internal/loader/ModuleRequest.js

```diff
@@ -36,24 +38,39 @@ loaders.set('esm', async (url) => {
 
 // Strategy for loading a node-style CommonJS module
 loaders.set('cjs', async (url) => {
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading CJSModule ${url}`);
-    const CJSModule = require('module');
-    const pathname = internalURLModule.getPathFromURL(new URL(url));
-    CJSModule._load(pathname);
+  debug(`Loading CJSModule ${url}`);
+  const CJSModule = require('module');
+  const pathname = internalURLModule.getPathFromURL(new URL(url));
+  const exports = CJSModule._load(pathname);
+  const es = exports[esModuleInterop] !== undefined ?
+    exports[esModuleInterop] : exports.__esModule;
+  const keys = es ? Object.keys(exports) : ['default'];
+  return createDynamicModule(keys, url, (reflect) => {
```

Comment:
Looks like `createDynamicModule` would throw if one of the exports was named `executor` because there would be duplicated exports? Seems like ideally it would generate an executor name that wouldn't conflict, or at least make it less likely to conflict.

Reply:
It maps names to safeguard against this already: https://github.com/nodejs/node/blob/f5a7a9e77d86e692f4bfce52fce2415e9235212d/lib/internal/loader/ModuleWrap.js#L18

Reply:
Ah so it does, my mistake. Missed the `$` on there.